qutip/tests/core/test_coefficient.py

```diff
@@ -34,8 +34,15 @@
 import pickle
 import qutip as qt
 import numpy as np
+from functools import partial
 from qutip.core.coefficient import (coefficient, norm, conj, shift,
-                                    reduce, CompilationOptions)
+                                    reduce, CompilationOptions,
+                                    clean_compiled_coefficient
+                                   )
+
+
+# Ensure the latest version is tested
+clean_compiled_coefficient(True)
```

Comment:
We shouldn't call clean up on import (otherwise any code that imports this test file will clear the cache). Instead we could add it to a `setup_function` function (see https://docs.pytest.org/en/stable/xunit_setup.html). This would also mean that the cache would be cleared between each test which would improve the isolation of the tests from each other.